client-runtime/src/main/java/com/microsoft/rest/v2/RestProxy.java

```diff
@@ -313,17 +313,34 @@ private Exception instantiateUnexpectedException(SwaggerMethodParser methodParse
         final Class<? extends RestException> exceptionType = methodParser.exceptionType();
         final Class<?> exceptionBodyType = methodParser.exceptionBodyType();
 
+        String bodyRepresentation;
+        if ("octet-stream".equalsIgnoreCase(response.headerValue("Content-Type"))) {
+            bodyRepresentation = "(" + response.headerValue("Content-Length") + "-byte body)";
+        } else {
+            bodyRepresentation = responseContent.isEmpty() ? "(empty body)" : "\"" + responseContent + "\"";
+        }
+
         Exception result;
         try {
             final Constructor<? extends RestException> exceptionConstructor = exceptionType.getConstructor(String.class, HttpResponse.class, exceptionBodyType);
-            final Object exceptionBody = responseContent == null || responseContent.isEmpty() ? null : serializer.deserialize(responseContent, exceptionBodyType, bodyEncoding(response.headers()));
 
-            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + responseContent, response, exceptionBody);
+            String contentType = response.headerValue("Content-Type");
+            boolean isSerializableContentType = contentType == null || contentType.isEmpty()
+                    || "application/json".equalsIgnoreCase(contentType)
+                    || "text/json".equalsIgnoreCase(contentType)
+                    || "application/xml".equalsIgnoreCase(contentType)
+                    || "text/xml".equalsIgnoreCase(contentType);
```

Comment:
Should these be `startsWith()` instead of `equals()` because there could be a `charset` value?

Reply:
Good question--probably should add coverage for such a contentType value, and add coverage for some non-serialized content type like text/html or something.